src/jit/lower.cpp

```diff
@@ -1480,6 +1479,10 @@ void Lowering::LowerArg(GenTreeCall* call, GenTreePtr* ppArg)
             BlockRange().Remove(arg);
             *ppArg = putArg;
         }
+        else
```

Comment:
where does this change come from?

Reply:
I'll have to look into that. I had made a change to how the decomposition of long fields was done, but it turned out to add more complexity than it saved, so I undid it. This may be a remnant from that change. I *think* that the old assert is still valid, because if it is a long field, the `GT_FIELD_LIST` will have `TYP_INT` because its type is the type of the first field in the list.